src/redisearch_rs/rlookup/src/lookup.rs

```diff
@@ -709,11 +739,37 @@ impl<'list, 'a> Iterator for CursorMut<'list, 'a> {
     }
 }
 
+// ===== impl RLookup =====
+
+impl Default for RLookup<'_> {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl<'a> RLookup<'a> {
+    pub fn new() -> Self {
+        Self {
+            keys: KeyList::new(),
+            options: RLookupOptions::empty(),
+            index_spec_cache: None,
+        }
+    }
+
+    pub fn init(&mut self, spcache: IndexSpecCache) {
+        assert!(self.index_spec_cache.is_none());
```

Comment:
C implementation was not panicking/aborting on re-initialization:
https://github.com/RediSearch/RediSearch/blob/4070ade95a3006b4c5a607bf0a0a622edece0abf/src/rlookup.c#L302-L303
If we want to keep the same stricness, what about making it a `debug_assert` and letting the old value to be dropped?